### PR TITLE
Add Stop Jobs action

### PR DESCRIPTION
**Pull request by gabriels1234**
Adds #573, action to stop a job. Current behavior deletes the entry but does not stop the job. 
@selwin

### CODE REVIEW

**Comment by gabriels1234** [comment]
@selwin Please let me know if it needs any change.
Thanks!

**Comment by selwin** [comment]
Hey, I was thinking we should add the `Stop` button on job detail page like this, along with other buttons (screenshot attached).

CleanShot 2023-03-23 at 15 59 18

File: django_rq/views.py [hunk 1]
```diff
@@ -493,6 +494,17 @@ def actions(request, queue_index):
                 for job_id in job_ids:
                     requeue_job(job_id, connection=queue.connection)
                 messages.info(request, 'You have successfully requeued %d  jobs!' % len(job_ids))
+            elif request.POST['action'] == 'stop':
+                cancelled_jobs = 0
+                for job_id in job_ids:
+                    try:
+                        send_stop_job_command(queue.connection, job_id)
+                        job = Job.fetch(job_id, connection=queue.connection)
```
Comment (selwin) [inline django_rq/views.py:502]: You shouldn't need to manually cancel the job. Does RQ not already automatically put the job in `FailedJobRegistry`?

If that's not the case this should be a bug that needs to be fixed within RQ.

Comment (gabriels1234) [reply django_rq/views.py:502]: @selwin Well, I was struggling a bit with the tests (to make it actually catch the job before it's marked as finished). 
But also, if you stop the job, you might not want to have the job in the failed column. I will be doing some tests so I'm happy to hold a bit. Also I will add the stop button that you suggested.

File: django_rq/templates/django_rq/job_detail.html [hunk 2]
```diff
@@ -186,6 +186,12 @@
 
     <div class="submit-row">
         <p class="deletelink-box"><a href="delete/" class="deletelink">Delete</a></p>
+        {% if job.is_started %}
+            <form method = 'POST' action = "{% url 'rq_requeue_job' queue_index job.id %}">
```
Comment (selwin) [inline django_rq/templates/django_rq/job_detail.html:190]: This does not look like it points to the correct URL.

Comment (selwin) [reply django_rq/templates/django_rq/job_detail.html:190]: @gabriels1234 let me know if you can fix this, I'd like to merge this in time for the next release :)

File: django_rq/urls.py [hunk 3]
```diff
@@ -30,4 +30,7 @@
     re_path(
         r'^queues/(?P<queue_index>[\d]+)/(?P<job_id>[-\w\.\:\$]+)/enqueue/$', views.enqueue_job, name='rq_enqueue_job'
     ),
+    re_path(
+        r'^queues/(?P<queue_index>[\d]+)/(?P<job_id>[-\w\.\:\$]+)/stop/$', views.stop_job, name='rq_stop_job'
```
Comment (selwin) [inline django_rq/urls.py:34]: Mind updating the regex for `job_id` to match the newly updated one in this PR? https://github.com/rq/django-rq/pull/586/files

File: django_rq/views.py [hunk 4]
```diff
@@ -529,3 +536,20 @@ def enqueue_job(request, queue_index, job_id):
         'queue': queue,
     }
     return render(request, 'django_rq/delete_job.html', context_data)
+
+
+@never_cache
+@staff_member_required
+def stop_job(request, queue_index, job_id):
+    """Stop started job"""
+    if request.method == 'POST':
```
Comment (selwin) [inline django_rq/views.py:545]: We can use `require_POST` view decorator so we don't need to check for the method. https://docs.djangoproject.com/en/4.2/topics/http/decorators/#django.views.decorators.http.require_POST

File: django_rq/templates/django_rq/job_detail.html (hunk 2 above)
Comment (gabriels1234) [reply django_rq/templates/django_rq/job_detail.html]: Done.

File: django_rq/views.py (hunk 1 above)
Comment (gabriels1234) [reply django_rq/views.py]: @selwin I fixed as per your requests, stop job does not cancel it. it just goes to the Failed registry.
Also the button was added.
I think it's ready to be merged.

File: django_rq/urls.py (hunk 3 above)
Comment (gabriels1234) [reply django_rq/urls.py]: Done.

File: django_rq/views.py (hunk 4 above)
Comment (gabriels1234) [reply django_rq/views.py]: Done.

**Comment by gabriels1234** [comment]
@selwin let me know if more changes are needed for this to be merged.
Thanks!

**Comment by selwin** [comment]
Thanks!

**Pull request merged**